account_invoice_supplier_self_invoice/models/res_partner.py

```diff
@@ -0,0 +1,48 @@
+# © 2017 Creu Blanca
+# License AGPL-3.0 or later (https://www.gnuorg/licenses/agpl.html).
+
+from odoo import fields, models
+
+
+class ResPartner(models.Model):
+    _inherit = "res.partner"
+
+    self_invoice = fields.Boolean(
+        string="Approves self invoice",
+        default=False,
+        help="When checked, all vendor bills will generate by default "
+        "a self-invoice on confirmation.",
+        tracking=True,
+    )
+    self_invoice_sequence_id = fields.Many2one(
+        comodel_name="ir.sequence", string="Self invoice sequence", ondelete="restrict"
+    )
+    self_invoice_report_footer = fields.Text(
+        string="Self invoice footer",
+        help="Footer text displayed at the bottom of the self invoice reports.",
+    )
+
+    def set_self_invoice(self):
+        for record in self:
+            record.self_invoice = not record.self_invoice
+            if record.self_invoice_sequence_id:
+                continue
+            if record.self_invoice:
+                record.self_invoice_sequence_id = (
+                    self.env["ir.sequence"]
+                    .sudo()
+                    .create(
```

Comment:
I will add the vendor reference if the field is not empty. (partner.ref  on Sales and Purchases > Misc group)

Reply:
Partners have refs? Or are you talking about adding a new field for that? If so, maybe it would be better to use partner ID? Example:

    SI/P%(partner_id)s/%(year)s/ -> SI/P123/2022/00001

Reply:
- Sequences should be same length (Best practices) so "partner_id will change length
- Partners have ref, yes.

Could it be a magic to have always for example 6 digit in the id?

`SI/P%(partner_id)s/%(year)s/ -> SI/P123/2022/00001`

This we should think more about that in an internacnional enviroment with multi-company

Reply:
> Sequences should be same length (Best practices) so "partner_id will change length

In this case, it should not affect, because the partner ID is not part of the sequence, but it is part of the prefix.

In any case, I'm seeing that it wouldn't help, because Partner 123 is the same for company A and B. It should be the company prefix instead.

> Partners have ref, yes.

Where?